### PR TITLE
Add preferences for default light/dark mode themes

### PR DESCRIPTION
**Pull request by Reisse**
Hi! This is an attempt at implementing #65.

Preferences are injected into rendered Markdown page by markdown-it plugin, as configuration API is not directly accessible from the webview. For now, only Mermaid theme is configurable. If this approach looks OK for you, more preferences may be added in the future.

Required VS Code version bumped accordingly for configuration API.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -61,4 +72,15 @@ export function activate() {
...
         .replace(/\</g, '&lt;')
-        .replace(/\>/g, '&gt;');
\ No newline at end of file
+        .replace(/\>/g, '&gt;');
+
+function injectMermaidTheme(md) {
+    const render = md.renderer.render;
+    md.renderer.render = function() {
+        return `<span id="${configSection}"
+                    darkModeTheme="${vscode.workspace.getConfiguration(configSection).get('darkModeTheme')}"
```
Comment (mjbvz) [inline src/index.ts:81]: Make sure to sanitize the config values here. We should only allow the `enum` values listed in the `package.json`

Comment (mjbvz) [inline src/index.ts:81]: These should likely be [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes)

File: src/index.ts [hunk 2]
```diff
@@ -61,4 +72,15 @@ export function activate() {
...
     source
         .replace(/\</g, '&lt;')
-        .replace(/\>/g, '&gt;');
\ No newline at end of file
+        .replace(/\>/g, '&gt;');
+
+function injectMermaidTheme(md) {
+    const render = md.renderer.render;
+    md.renderer.render = function() {
+        return `<span id="${configSection}"
```
Comment (mjbvz) [inline src/index.ts:80]: Perhaps this should be a `<meta>` tag or at least set `aria-hidden` so it is not treated as part of the document for accessibility

**Review by mjbvz — CHANGES_REQUESTED** [review]
Looks like a good approach overall. Just a few minor comments about the implementation

File: src/index.ts (hunk 1 above)
Comment (Reisse) [reply src/index.ts]: Done!

Comment (Reisse) [reply src/index.ts]: Yes, thanks for the link!

File: src/index.ts (hunk 2 above)
Comment (Reisse) [reply src/index.ts]: I haven't found how to modify `<head>` part from the renderer, so just set `aria-hidden`.

**Comment by mjbvz** [comment]
Thank you! Will publish update with the fix shortly

**Pull request merged**